packages/beacon-node/src/network/peers/peerManager.ts

```diff
@@ -606,22 +606,18 @@ export class PeerManager {
       void this.requestStatus(remotePeer, this.statusCache.get());
     }
 
-    // AgentVersion was set in libp2p IdentifyService, 'peer:connect' event handler
-    // since it's not possible to handle it async, we have to wait for a while to set AgentVersion
-    // See https://github.com/libp2p/js-libp2p/pull/1168
-    retry(
-      async () => {
-        const agentVersionBytes = (await this.libp2p.peerStore.get(peerData.peerId)).metadata.get("AgentVersion");
-        if (agentVersionBytes) {
-          const agentVersion = new TextDecoder().decode(agentVersionBytes) || "N/A";
+    this.libp2p.services.identify
+      .identify(evt.detail)
+      .then((result) => {
+        const agentVersion = result.agentVersion;
+        if (agentVersion) {
```

Comment:
I am curios about the different cases, i.e. when do we not get an error but also an empty agentVersion? and should we add a fallback case like `"N/A"` as previously done?

Reply:
The protobuf sent to us can have a null entry.

I thought about keeping the fallback "N/A" but it didn't seem particularly useful? Happy to add it back if anyone feels different.

Reply:
> I thought about keeping the fallback "N/A" but it didn't seem particularly useful?

The only case I see were this makes a difference is in the case we error as it would be `undefined` vs `"N/A"` otherwise, although I don't see that this value is used anywhere other than dumping peer information.

I remember Tuyen did some changes in known client version but that is just for metrics afaik and we already handle the case if `agentClient` is not set

https://github.com/ChainSafe/lodestar/blob/3adf2c2eb233483b5bea2abf253a7cce68c36af9/packages/beacon-node/src/network/peers/peerManager.ts#L697